src/Clients/Consumer/KafkaConsumer.php

```diff
@@ -144,50 +145,47 @@ function (Message $message) use (
     }
 
     /**
-     * @param callable(Message) : void $onSuccess
-     * @param callable() : void        $onPartitionEof
-     * @param callable() : void        $onTimedOut
+     * @param callable(Message): void $onSuccess
+     * @param callable() : void       $onPartitionEof
+     * @param callable() : void       $onTimedOut
      */
     private function doStart(
         int $timeoutMs,
         callable $onSuccess,
         ?callable $onPartitionEof = null,
         ?callable $onTimedOut = null
     ): void {
-        $this->registerSignals();
+        $this->registerSignals($this->shouldRun);
 
-        try {
-            while (true) {
-                pcntl_signal_dispatch();
+        while ($this->shouldRun) {
+            pcntl_signal_dispatch();
```

Comment:
I'd move this bellow the switch

Reply:
Why? TBH I can't decide and found nothing on the topic.

Reply:
Since the callback may affect the loop it makes sense to me to be "just before the loop condition evaluation".

Reply:
Sounds legit and we have nothing better anyway 😄 . Moved